src/scenic/core/simulators.py

```diff
@@ -353,9 +354,9 @@ def currentState(self):
         The definition of 'state' is up to the simulator; the 'state' is simply saved
         at each time step to define the 'trajectory' of the simulation.
 
-        The default implementation returns a tuple of the positions of all objects.
+        The default implementation a list of objects with their properties.
         """
-        return tuple(obj.position for obj in self.objects)
+        return [self.getProperties(obj, obj.properties) for obj in self.objects]
```

Comment:
We don't want to call `getProperties` again since it may be slow (e.g. actually talking to the simulator). I would just return `{prop: getattr(obj, prop) for prop in obj._dynamicProperties}`. Also the docstring should clarify that it's only *dynamic* properties that are saved (you can link to the glossary by writing ":term:\`dynamic properties\`").